assets/js/base/context/hooks/cart/use-store-cart.ts

```diff
@@ -175,8 +175,16 @@ export const useStoreCart = (
 				decodeValues( fee )
 			);
 
+			// Add a text property to the coupon to allow extensions to modify
+			// the text used to display the coupon, without affecting the
+			// functionality when it comes to removing the coupon.
+			const cartCoupons = cartData.coupons.map( ( coupon ) => ( {
+				...coupon,
+				text: coupon.code,
```

Comment:
Wondering if `label` would be more appropriate than `text`. For example, we have the Total label that the user can change, so to keep it in the same registry I'd say a `coupon label` makes it more clear that we are talking about what text is displayed. What do you think?